src/lib.rs

```diff
@@ -42,10 +43,37 @@ pub enum KVMiInterceptType {
     Breakpoint = KVMI_EVENT_BREAKPOINT as isize,
     Pagefault = KVMI_EVENT_PF as isize,
 }
+
+#[derive(Debug, Copy, Clone)]
+#[repr(u8)]
 pub enum KVMiPageAccess {
-    PageAccessW = kvmi_sys::KVMI_PAGE_ACCESS_W as isize,
-    PageAccessR = kvmi_sys::KVMI_PAGE_ACCESS_R as isize,
-    PageAccessX = kvmi_sys::KVMI_PAGE_ACCESS_X as isize,
+    W = kvmi_sys::KVMI_PAGE_ACCESS_W as u8,
+    R = kvmi_sys::KVMI_PAGE_ACCESS_R as u8,
+    X = kvmi_sys::KVMI_PAGE_ACCESS_X as u8,
+    NIL = 0,
```

Comment:
mini nitpick: Just a suggestion but find it really helpful to order integer enums in ascending order.